src/methods/equals.jl

```diff
@@ -0,0 +1,192 @@
+
+First, we implement a wrapper method that dispatches to the correct
+implementation based on the geometry trait. This is also used in the
+implementation, since it's a lot less work! 
+
+Note that while we need the same set of points and edges, they don't need to be
+provided in the same order for polygons. For for example, we need the same set
+points for two multipoints to be equal, but they don't have to be saved in the
+same order. This requires checking every point against every other point in the
+two geometries we are comparing.  
+=#
+
+"""
+    equals(geom1, geom2)::Bool
+
+Compare two Geometries return true if they are the same geometry.
+
+## Examples
+```jldoctest
+import GeometryOps as GO, GeoInterface as GI
+poly1 = GI.Polygon([[(0,0), (0,5), (5,5), (5,0), (0,0)]])
+poly2 = GI.Polygon([[(0,0), (0,5), (5,5), (5,0), (0,0)]])
+
+GO.equals(poly1, poly2)
+# output
+true
+```
+"""
+equals(geom_a, geom_b) = equals(
+    GI.trait(geom_a), geom_a,
+    GI.trait(geom_b), geom_b,
+)
+
+"""
+    equals(::T, geom_a, ::T, geom_b)::Bool
+
+Two geometries of the same type, which don't have a equals function to dispatch
+off of should throw an error.
+"""
+equals(::T, geom_a, ::T, geom_b) where T = error("Cant compare $T yet")
+
+"""
+    equals(trait_a, geom_a, trait_b, geom_b)
+
+Two geometries which are not of the same type cannot be equal so they always
+return false.
+"""
+equals(trait_a, geom_a, trait_b, geom_b) = false
+
+"""
+    equals(::GI.PointTrait, p1, ::GI.PointTrait, p2)::Bool
+
+Two points are the same if they have the same x and y (and z if 3D) coordinates.
+"""
+function equals(::GI.PointTrait, p1, ::GI.PointTrait, p2)
+    GI.ncoord(p1) == GI.ncoord(p2) || return false
+    GI.x(p1) == GI.x(p2) || return false
+    GI.y(p1) == GI.y(p2) || return false
+    if GI.is3d(p1)
+        GI.z(p1) == GI.z(p2) || return false 
+    end
+    return true
+end
+
+"""
+    equals(::GI.MultiPointTrait, mp1, ::GI.MultiPointTrait, mp2)::Bool
+
+Two multipoints are equal if they share the same set of points.
+"""
+function equals(::GI.MultiPointTrait, mp1, ::GI.MultiPointTrait, mp2)
+    GI.npoint(mp1) == GI.npoint(mp2) || return false
+    for p1 in GI.getpoint(mp1)
+        has_match = false  # if point has a matching point in other multipoint
+        for p2 in GI.getpoint(mp2)
+            if equals(p1, p2)
+                has_match = true
+                break
+            end
+        end
+        has_match || return false  # if no matching point, can't be equal
+    end
+    return true  # all points had a match
+end
+
+"""
+    equals(::T, l1, ::T, l2) where {T<:GI.AbstractCurveTrait} ::Bool
+
+Two curves are equal if they share the same set of points going around the
+curve. 
+"""
+function equals(::T, l1, ::T, l2) where {T<:GI.AbstractCurveTrait}
+    # Check line lengths match
+    n1 = GI.npoint(l1)
+    n2 = GI.npoint(l2)
+    # TODO: do we need to account for repeated last point??
```

Comment:
If there are two polygons, one with a repeated first point at the end, and one without the repeated point, should they still be equivalent?

Reply:
I think they should be. A polygon is closed by definition, so that is just a matter of representation if the last point is repeated.

Reply:
Thoughts on this?

Reply:
I updated this! It now works. Hopefully, my implementation isn't too complicated. I will push soon.

Reply:
Sorry forgot to reple. Its harder for GeoInterface because Geome6ryBasics uses Linestring for everything so ww can enforce these things.

Glad you found a solution